airbyte-cdk/bulk/toolkits/load-iceberg-parquet/src/main/kotlin/io/airbyte/cdk/load/toolkits/iceberg/parquet/IcebergTableSynchronizer.kt

```diff
@@ -174,6 +194,32 @@ class IcebergTableSynchronizer(
             update.setIdentifierFields(updatedIdentifierFields)
         }
 
+        // If we're doing separate commits for column replacements, commit the delete operations now
+        if (requireSeparateCommitsForColumnReplace && columnsToReplaceInSecondCommit.isNotEmpty()) {
+            // Commit the first update (with deletes but not adds for replaced columns)
+            update.commit()
```

Comment:
should we be checking `columnTypeChangeBehavior.commitImmediately` here? feels weird to unconditionally commit a schema change when we have a parameter that claims to control it

(or if this is a safe thing to commit immediately - then we should rename that param to something less misleading)